crates/bevy_core_pipeline/src/motion_blur/motion_blur.wgsl

```diff
@@ -0,0 +1,168 @@
+    for (var i = -n_samples; i < n_samples; i++) {
+        // The current sample step vector, from in.uv
+        let step_vector = 0.5 * exposure_vector * (f32(i) + noise) / f32(n_samples);
+        var sample_uv = in.uv + step_vector;
+        let sample_coords = vec2<i32>(sample_uv * texture_size);
+
+    #ifdef MULTISAMPLED
+        let sample_color = textureLoad(screen_texture, sample_coords, i32(sample_index));
+    #else
+        let sample_color = textureSample(screen_texture, texture_sampler, sample_uv);
+    #endif
+    #ifdef MULTISAMPLED
+        let sample_motion = textureLoad(motion_vectors, sample_coords, i32(sample_index)).rg;
+    #else
+        let sample_motion = textureSample(motion_vectors, texture_sampler, sample_uv).rg;
+    #endif
+    #ifdef NO_DEPTH_TEXTURE_SUPPORT
+        let sample_depth = 0.0;
+    #else
+    #ifdef MULTISAMPLED
+        let sample_depth = textureLoad(depth, sample_coords, i32(sample_index));
+    #else
+        let sample_depth = textureSample(depth, texture_sampler, sample_uv);
+    #endif
+    #endif
+
+        var weight = 1.0;
+        let is_sample_in_fg = !(depth_supported && sample_depth < this_depth && sample_depth > 0.0);
+        if is_sample_in_fg {
+            // The following weight calculation is used to eliminate ghosting artifacts that are
+            // common in motion-vector-based motion blur implementations. While some resources
+            // recommend using depth, I've found that sampling the velocity results in significantly
+            // better results. Unlike a depth heuristic, this is not scale dependent.
+            //
+            // The most distracting artifacts occur when a stationary foreground object is
+            // incorrectly sampled while blurring a moving background object, causing the stationary
+            // object to blur when it should be sharp ("background bleeding"). This is most obvious
+            // when the camera is tracking a fast moving object. The tracked object should be sharp,
+            // and should not bleed into the motion blurred background.
+            //
+            // To attenuate these incorrect samples, we compare the motion of the fragment being
+            // blurred to the UV being sampled, to answer the question "is it possible that this
+            // sample was occluding the fragment?"
+            //
+            // Note to future maintainers: proceed with caution when making any changes here, and
+            // ensure you check all oclusion/disocclusion scenarios and fullscreen camera rotation
+            // blur for regressions.
+            let frag_speed = length(step_vector);
+            let sample_speed = length(sample_motion) / 2.0; // Halved because the sample is centered
+            let cos_angle = dot(step_vector, sample_motion) / (frag_speed * sample_speed * 2.0);
+            let motion_similarity = clamp(abs(cos_angle), 0.0, 1.0);
+            if sample_speed * motion_similarity < frag_speed {
```

Comment:
Same thing here, turn an if statement into a weight using f32(condition) and multiply to avoid the branch.

Reply:
Is that actually going to make any difference? This should optimize away completely. I'd think we would want to prioritize readability in cases like this where there should be no perf impact.

Reply:
I don't think this `if` will be a branch. If we're really worried we could use a select. (or check the code gen) I think the cast to f32 and multiply would definitely be more ops.

Reply:
Makes sense